pandas/tests/indexes/test_base.py

```diff
@@ -302,138 +291,138 @@ def test_index_ctor_infer_periodindex(self):
-                           pd.date_range('2011-01-01', periods=5).asi8]:
+        res = klass(values, tz=tz)
+        tm.assert_index_equal(res, idx)
 
-                for res in [pd.Index(values, tz=tz),
-                            pd.Index(values, dtype=dtype),
-                            pd.Index(list(values), tz=tz),
-                            pd.Index(list(values), dtype=dtype)]:
-                    tm.assert_index_equal(res, idx)
+        res = klass(values, dtype=dtype)
+        tm.assert_index_equal(res, idx)
 
-                # check compat with DatetimeIndex
-                for res in [pd.DatetimeIndex(values, tz=tz),
-                            pd.DatetimeIndex(values, dtype=dtype),
-                            pd.DatetimeIndex(list(values), tz=tz),
-                            pd.DatetimeIndex(list(values), dtype=dtype)]:
-                    tm.assert_index_equal(res, idx)
+        res = klass(list(values), tz=tz)
+        tm.assert_index_equal(res, idx)
 
-    def test_constructor_dtypes_timedelta(self):
+        res = klass(list(values), dtype=dtype)
+        tm.assert_index_equal(res, idx)
 
+    @pytest.mark.parametrize("attr", ['values', 'asi8'])
+    @pytest.mark.parametrize("klass", [pd.Index, pd.TimedeltaIndex])
+    def test_constructor_dtypes_timedelta(self, attr, klass):
         idx = pd.timedelta_range('1 days', periods=5)
         dtype = idx.dtype
 
-        for values in [idx.values, idx.asi8]:
+        values = getattr(idx, attr)
 
-            for res in [pd.Index(values, dtype=dtype),
-                        pd.Index(list(values), dtype=dtype)]:
-                tm.assert_index_equal(res, idx)
+        res = klass(values, dtype=dtype)
+        tm.assert_index_equal(res, idx)
 
-            # check compat with TimedeltaIndex
-            for res in [pd.TimedeltaIndex(values, dtype=dtype),
-                        pd.TimedeltaIndex(list(values), dtype=dtype)]:
-                tm.assert_index_equal(res, idx)
+        res = klass(list(values), dtype=dtype)
+        tm.assert_index_equal(res, idx)
 
-    def test_constructor_empty(self):
+    def test_constructor_empty_gen(self):
         skip_index_keys = ["repeats", "periodIndex", "rangeIndex",
                            "tuples"]
         for key, idx in self.generate_index_types(skip_index_keys):
             empty = idx.__class__([])
             assert isinstance(empty, idx.__class__)
             assert not len(empty)
 
-        empty = PeriodIndex([], freq='B')
-        assert isinstance(empty, PeriodIndex)
-        assert not len(empty)
-
-        empty = RangeIndex(step=1)
-        assert isinstance(empty, pd.RangeIndex)
-        assert not len(empty)
-
-        empty = MultiIndex(levels=[[1, 2], ['blue', 'red']],
-                           labels=[[], []])
-        assert isinstance(empty, MultiIndex)
+    @pytest.mark.parametrize("empty,klass", [
+        (PeriodIndex([], freq='B'), PeriodIndex),
```

Comment:
might be duplicated elsewhere. am iffy on where to put things like this. maybe worth extracting things from this file and makign a ``test_construction`` for things like this